### ISSUE TITLE
i386 build for Debian depends on wrong version GLIBC_2.27 instead of GLIBC_2.24

### ISSUE THREAD

**Issue by getreu** [wireapp/wire-desktop#2038]
I installed the debian i386 version of `wire-desktop` from your repository `deb https://wire-app.wire.com/linux/debian stable main`. The installation went fine and installed `wire-desktop` version:  3.5.2881-2881.

When I start the binary I get the following error:

    wire-desktop:/lib/i386-linux-gnu/libc.so.6: version GLIBC_2.27 not found

This is a bug, because stable Debian Stretch applications should link with `GLIBC_2.24` instead.

**Comment by lbernd** [comment]
Same error message with lubuntu 16.04.1 i386 and wire-desktop 3.5.2881-2881 installed from repository. Available GLIBC version is 2.23 (Ubuntu GLIBC 2.23-0ubuntu10). Error message gone with wire-desktop version 3.4.2879-287.

Than you in advance for any information. Please tell me If I should have opened another issue. Thank you for developing wire!

**Comment by hawkins-dale** [comment]
My Wire install stopped working on Linux Mint (14.04.5 LTS, Trusty Tahr).  I reinstalled, and now I get the same message.

Hope this helps!

**Comment by ebym** [comment]
Same GLIBC problem with Wire 3.5.2881 on Mint 17.3 (Trusty), reverting back to old version 3.0.2816, 32bit gives white screen with a side bar. Synaptic do not show any other versions other than the latest and 3.0.2816 !!!.
However the bowser version just works (FF 63).

**Comment by ffflorian** [comment]
Bad news: Electron will [discontinue the support for Linux 32-bit](https://electronjs.org/blog/linux-32bit-support), so this is unlikely to be fixed.

**Comment by strypey** [comment]
@ffflorian It says at that link that 32-bit will be support for Electron 3.1 until the release of Electron 6. Electron 4 came out 2 months ago, and Electron 3 was released four months before that. So that mean we've probably got at least 6 months of 32-bit support on Electron 3.1. Would it be possible to base 32-bit Wire-desktop off 3.1 (if it isn't already), and get this bug fixed, to give us all a few months of runway to figure out what our options are?

**Comment by ffflorian** [comment]
@strypey 
> Would it be possible to base 32-bit Wire-desktop off 3.1 (if it isn't already), and get this bug fixed, to give us all a few months of runway to figure out what our options are?

I see your point, but as stated [in our Wiki regarding Linux](https://github.com/wireapp/wire-desktop/wiki/How-to-install-Wire-for-Desktop-on-Linux):
> We don't offer official support at the moment [...]

So unfortunately we can't offer additional support for 32-bit Linux.

**Comment by strypey** [comment]
Understood. But if somebody in the wider Wire community could come up with a fix along those lines, would a PR get merged, and a working version released?

**Comment by hawkins-dale** [comment]
As @ebym points out:  the web version, for me, Just Works.

**Comment by strypey** [comment]
@hawkins-dale 
Including voice calls?

**Comment by hawkins-dale** [comment]
@strypey :  Oh!  No, I have never tried voice on the web client.  I just use Wire on my phone for voice.  
Good point.

**Comment by ebym** [comment]
I can confirm desktop version 3.4.2879 is working on 32bit linux, you can download it from project page and install manually, never do apt update. 
But still don't know how long it would work for.

@strypey & @hawkins-dale i can confirm video and voice calls work with the web client. Mint 17.3 (Trusty) & FF63.

**Comment by bennyn** [comment]
> But if somebody in the wider Wire community could come up with a fix along those lines, would a PR get merged, and a working version released?

How would such a fix look like? Do you have a solution in mind? Our desktop application is dependant on Electron and since Electron discontinued the support for Linux 32-bit, we have to discontinue it too. 

⚠️ **Info:** We recommend everyone who wants to continue using Wire on Linux (32-bit) to convert to [our web application](https://app.wire.com/). You can [backup your conversation history](https://support.wire.com/hc/en-us/articles/360000775549) and restore it in our web client.  

Audio and video calls are also supported by our web version. On Firefox we [support screensharing](https://support.wire.com/hc/en-us/articles/208958069) too.

**Comment by ebym** [comment]
@bennyn > since Electron discontinued the support for Linux 32-bit, we have to discontinue it too.

Any tentative date for this ?.

**Comment by strypey** [comment]
@bennyn 
> Do you have a solution in mind?

Distributing Wire 3.4.2879 to users with 32-bit GNU/Linux systems, instead of the newer version that doesn't work, would be a quick fix.

> since Electron discontinued the support for Linux 32-bit, we have to discontinue it too.

Understood, but not until Electron 6.0 comes out, which as I say, is months away, and for all we know may not happen until next year. If you can offer a working version for 32-bit systems until then, the community could look into other options, like supporting dev on a non-Electron client (worst-case-scenario switching to another app or buying new hardware).

> Audio and video calls are also supported by our web version.

I'll test this again, but it wasn't the case on my system the last time I tried. If it still doesn't work, I'll report it as a new issue.

**Comment by ffflorian** [comment]
> Any tentative date for this ?

The last version [3.9.2895](https://github.com/wireapp/wire-desktop/releases/tag/linux%2F3.9.2895) was the first version released only as 64-bit.

**Comment by ffflorian** [comment]
> If you can offer a working version for 32-bit systems until then, the community could look into other options

Please note that the Linux version is still experimental and is not officially supported.

Closing this since the 32-bit version is no longer supported.

**Comment by strypey** [comment]
@ffflorian
> Please note that the Linux version is still experimental and is not officially supported.

Fair point, and I accept also that many of us filing bugs and making feature requests are not paying customers. The work you are doing to make the Wire service available to GNU/Linux users is appreciated :)
